transforms/listToArray.spec.js

```diff
@@ -0,0 +1,75 @@
+const test = require('tape')
+const helpers = require('../test/helpers')
+
+const bindFunc = helpers.bindFunc
+
+const constant = require('../combinators/constant')
+const identity = require('../combinators/identity')
+
+const isFunction = require('../predicates/isFunction')
+const isArray = require('../predicates/isArray')
+const isSameType = require('../predicates/isSameType')
+
+const List = require('../crocks/List')
+
+const listToArray = require('./listToArray')
+
+test('listToArray transform', t => {
+  const f = bindFunc(listToArray)
+
+  t.ok(isFunction(listToArray), 'is a function')
+
+  t.throws(f(undefined), TypeError, 'throws if arg is undefined')
+  t.throws(f(null), TypeError, 'throws if arg is null')
+  t.throws(f(0), TypeError, 'throws if arg is a falsey number')
+  t.throws(f(1), TypeError, 'throws if arg is a truthy number')
+  t.throws(f(''), TypeError, 'throws if arg is a falsey string')
+  t.throws(f('string'), TypeError, 'throws if arg is a truthy string')
+  t.throws(f(false), TypeError, 'throws if arg is false')
+  t.throws(f(true), TypeError, 'throws if arg is true')
+  t.throws(f({}), TypeError, 'throws if arg is an object')
+
+  t.end()
+})
+
+test('listToArray with List', t => {
+  const data = [23, 45, 'a', [42, 56, ['19']]]
+  const list = List.fromArray(data)
+
+  const a = listToArray(list)
+
+  t.ok(isArray(a), 'returns an Array from List')
+
+  t.end()
+})
+
+test('listToArray with List returning function', t => {
+  const f = bindFunc(listToArray(identity))
+
+  t.throws(f(undefined), TypeError, 'throws if function returns undefined')
+  t.throws(f(null), TypeError, 'throws if function returns null')
+  t.throws(f(0), TypeError, 'throws if function returns a falsey number')
+  t.throws(f(1), TypeError, 'throws if function returns a truthy number')
+  t.throws(f(''), TypeError, 'throws if function returns a falsey string')
+  t.throws(f('string'), TypeError, 'throws if function returns a truthy string')
+  t.throws(f(false), TypeError, 'throws if function returns false')
+  t.throws(f(true), TypeError, 'throws if function returns true')
+  t.throws(f({}), TypeError, 'throws if function returns an object')
+  t.throws(f([]), TypeError, 'throws if function returns an array')
+
+  const data = [32, 54, 'b', [24, 65], ['91']]
+  const list = List.fromArray(data)
+
+  const k = constant(list)
+  const l = k(listToArray)
+  const m = listToArray(identity, list)
+
+  t.ok(isSameType(List, l), 'returns a function which returns a List')
```

Comment:
This is not the unit under test.
No need to test this. This really tests nothing here.

remember that `constant` is defined: `a -> _ -> a`, so `k(listToArray)` is just ignoring `listToArray` and just returning the List that was passed to constant above. Please remove ALL of this. No need to do all this setup or test your setup.